tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala

```diff
@@ -904,4 +904,37 @@ class MinorDottySuite extends BaseDottySuite {
       )
     )
   }
+  test("regex-pattern") {
+    runTestAssert[Stat](
+      """|s match
+         |  case re(v): String => v.toDouble
```

Comment:
Probably, it's needed to remove all checks from `Typed.lhs`

I tried the following code and dotty compiles it:
```scala
s match {
  case (v: String): String =>  // <- scalameta tree `Pat.Typed(Pat.Typed(...), ...)`
}
```

Reply:
Good catch, I removed all of them and also added another test.